### DISCUSSION TITLE
Definition and Hover

### DISCUSSION THREAD

**Discussion by SoilihBenSoilih** [c-cube/linol#9] [General]
Hello,
Hope you are good !
It's me again haha. I'm sorry to disturb you too much.
I'm still developping my lsp server using Linol for Typer
(a programming language). I have implemented the 
methods on_req_hover and on_req_definition( not final),
It was for "testing" to see how it works and in emacs, it says
` The connected server(s) does not support method textDocument/definition`.
And it is the same with the hover. Have I something I did wrong?
I'm using the version 0.2 of Linol. 
my code is here : [typer-lsp-server.txt](https://github.com/c-cube/linol/files/6361544/typer-lsp-server.txt)

Just the methods on_req_definition and on_req_hover or one of them.

Thank you a lot !!!

**Comment by c-cube** [comment]
Which editor are you testing with? I have the impression that VSCode, for example, is very compliant and only calls methods that have been declared in the capabilities.

On the main branch I changed a lot of things, including new methods `config_xxx` to declare particular capabilities (e.g. `method config_definition = `Bool true` and `method config_hover = `Bool true` would help in your case if it's a capabilities problem). However some methods' signatures have changed (with a `notify_back` additional argument).

The master branch also uses `logs` as a dependency so that you can log queries and replies (you do need to setup a log reporter though). Sorry about the unstability, these are pretty early days.

**Reply by SoilihBenSoilih** [reply]
Hello,

I use emacs as an editor. Oh I see, i will try it.
For the logs, do you have an exemple somewhere
where you are doing that. Thank you another time 
for this library, it's really helpfull, I am still a student
and it's my graduation project I'm on. Sorry if I ask
you a lot of questions !

Thank you another time !

**Reply by c-cube** [reply]
for logs: https://erratique.ch/software/logs/doc/Logs/index.html#setupreporter (I don't have an example at hand)

**Reply by SoilihBenSoilih** [reply]
Thank you !!!

**Reply by SoilihBenSoilih** [reply]
Hello, 
Hope you're doing well !
It is just to confirm that the set of configuration
to true fixed the problems in case there is someone
else with the same problem.

Thank you a lot again !
